server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/GlobalRateLimitIntegrationTest.java

```diff
@@ -0,0 +1,388 @@
+            .build();
+    }
+
+    @AfterEach
+    void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    void sendMessageShouldSucceedWhenCountLimitAcceptable() throws Exception {
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(SENDER1, PASSWORD)
+            .sendMessage(FakeMail.builder()
+                .name("name")
+                .mimeMessage(message)
+                .sender(SENDER1)
+                .recipient(RECIPIENT1));
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(SENDER2, PASSWORD)
+            .sendMessage(FakeMail.builder()
+                .name("name")
+                .mimeMessage(message)
+                .sender(SENDER2)
+                .recipient(RECIPIENT1));
+
+        testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(RECIPIENT1, PASSWORD)
+            .select(TestIMAPClient.INBOX)
+            .awaitMessageCount(awaitAtMostOneMinute, 2);
+    }
+
+    @Test
+    void sendMessageShouldFailWhenCountLimitExceeded() throws Exception {
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(SENDER1, PASSWORD)
+            .sendMessage(FakeMail.builder()
+                .name("name")
+                .mimeMessage(message)
+                .sender(SENDER1)
+                .recipient(RECIPIENT1));
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(SENDER2, PASSWORD)
+            .sendMessage(FakeMail.builder()
+                .name("name")
+                .mimeMessage(message)
+                .sender(SENDER2)
+                .recipient(RECIPIENT1));
+
+        // await first 2 message
+        testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(RECIPIENT1, PASSWORD)
+            .select(TestIMAPClient.INBOX)
+            .awaitMessageCount(awaitAtMostOneMinute, 2);
+
+        // send third message should exceed count limit
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(SENDER1, PASSWORD)
+            .sendMessage(FakeMail.builder()
+                .name("name")
+                .mimeMessage(message)
+                .sender(SENDER1)
+                .recipient(RECIPIENT2));
+
+        awaitAtMostOneMinute.untilAsserted(() -> jamesServer.getProbe(SpoolerProbe.class).processingFinished());
+
+        assertThat(testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(RECIPIENT2, PASSWORD)
+            .select(TestIMAPClient.INBOX)
+            .getMessageCount(TestIMAPClient.INBOX)).isEqualTo(0);
```

Comment:
Where is the mail? Can we have an assert on that? 

Hint: in `var/mail/error` mail repository